### PR TITLE
Better rate limited message.

### PR DESCRIPTION
**Pull request by vmihailenco**
Updates https://github.com/go-redis/redis/issues/222

@dim PTAL

### CODE REVIEW

File: multi.go [hunk 1]
```diff
@@ -46,16 +46,15 @@ func (c *Client) Multi() *Multi {
...
 		// Close current connection.
-		c.base.connPool.(*stickyConnPool).Reset()
+		c.base.connPool.(*stickyConnPool).Reset(err)
 	} else {
-		err = c.base.connPool.Put(cn)
-	}
-	if err != nil {
-		log.Printf("redis: putConn failed: %s", err)
+		err := c.base.connPool.Put(cn)
+		if err != nil {
```
Comment (dim) [inline multi.go]: gofmt?

File: pool.go [hunk 2]
```diff
@@ -204,15 +204,15 @@ func (p *connPool) wait() *conn {
...
+			"redis: you open connections too fast (last_error=%q)",
+			p.loadLastErr(),
 		)
 		return nil, err
 	}
 
 	cn, err := p.dialer()
 	if err != nil {
-		p.lastDialErr = err
+		p.storeLastErr(err.Error())
```
Comment (dim) [inline pool.go]: Wouldn't it make sense to store the full `error` and not just a `string`?

File: pool.go [hunk 3]
```diff
@@ -330,6 +333,17 @@ func (p *connPool) reaper() {
 	}
 }
 
+func (p *connPool) storeLastErr(err string) {
+	p.lastErr.Store(err)
+}
+
+func (p *connPool) loadLastErr() string {
```
Comment (dim) [inline pool.go]: And return `error` here, instead of `string` too?

**Comment by dim** [comment]
LGTM overall, see few notes above. thanks, dim

File: pool.go (hunk 2 above)
Comment (vmihailenco) [inline pool.go]: `atomic.Value` only allows to store values of one type and `error` interface can have different underlying types.

File: multi.go (hunk 1 above)
Comment (vmihailenco) [inline multi.go]: This is gofmted...

**Pull request merged**